spring-integration-core/src/main/java/org/springframework/integration/IntegrationMessageHeaderAccessor.java

```diff
@@ -157,4 +157,29 @@ protected boolean isReadOnly(String headerName) {
 		return super.isReadOnly(headerName) || this.readOnlyHeaders.contains(headerName);
 	}
 
+	public boolean containsReadOnly(MessageHeaders headers) {
```

Comment:
M-m-m. Looking to the usage of this method in the `MessageBuilder` I think we don't need it.
It is a bit confusing as public API.
The `IntegrationMessageHeaderAccessor` represents some volatile object around particular headers.
That smells a bit to check the state of this object against arbitrary external `MessageHeaders` instance.

WDYT?

I can fix on merge